src/Microsoft.CodeQuality.Analyzers/Core/QualityGuidelines/AbstractRemoveEmptyFinalizers.cs

```diff
@@ -35,72 +33,26 @@ public override void Initialize(AnalysisContext analysisContext)
             analysisContext.EnableConcurrentExecution();
             analysisContext.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
 
-            analysisContext.RegisterCompilationStartAction(compilationContext =>
+            analysisContext.RegisterSymbolAction(methodContext =>
```

Comment:
Because symbol seemed to me to be more natural and correct here. We're not looking at every code block, we're looking at every symbol declaration to see the body of the symbol. If we went with code block, then the first thing we'd have to do is get the symbol for that code block to see if it's an analyzer.